dev/com.ibm.ws.cdi.ejb.common/src/com/ibm/ws/cdi/ejb/impl/WebSphereEjbServicesImpl.java

```diff
@@ -60,29 +70,33 @@ public void registerInterceptors(EjbDescriptor<?> ejbDescriptor, InterceptorBind
                 }
 
                 EjbDescriptor<?> descriptor = ejbDescriptor;
-                //TODO in Weld 3, InternalEjbDescriptor is package protected and we can't use it
-//                if (ejbDescriptor instanceof InternalEjbDescriptor) {
-//                    InternalEjbDescriptor<?> internal = (InternalEjbDescriptor<?>) ejbDescriptor;
-//                    descriptor = internal.delegate();
-//                }
-                WebSphereEjbDescriptor<?> webSphereEjbDescriptor = (WebSphereEjbDescriptor<?>) descriptor;
+                WebSphereEjbDescriptor<?> webSphereEjbDescriptor = findWebSphereEjbDescriptor(descriptor);
                 J2EEName ejbJ2EEName = webSphereEjbDescriptor.getEjbJ2EEName();
 
                 interceptorRegistry.registerInterceptors(ejbJ2EEName, interceptorBindings);
             }
         }
     }
 
+    private WebSphereEjbDescriptor<?> findWebSphereEjbDescriptor(EjbDescriptor<?> descriptor) {
+        String j2eeName = EjbDescriptorImpl.parseJ2EEName(descriptor);
+        WebSphereEjbDescriptor<?> webSphereEjbDescriptor = null;
+        synchronized (ejbDescriptorMap) {
+            webSphereEjbDescriptor = ejbDescriptorMap.get(j2eeName);
+        }
+        if (webSphereEjbDescriptor == null || !descriptor.equals(webSphereEjbDescriptor)) {
+            //Shouldn't be possible to get here ... the ejbDescriptorMap is global so if the descriptor isn't in there then we didn't create it
+            //the one passed in should either be exactly the same as the one in the map or it should be one wrapped by Weld.
+            //If it was wrapped by Weld then the equals method should still defer down to our original and the two should be "equal".
+            throw new IllegalStateException();
```

Comment:
need to have more message on this.

Reply:
internal error one

Reply:
Yeah I know ... I wanted a functional review before worrying about NLS.